src/map.rs

```diff
@@ -89,6 +90,64 @@ pub struct Message {
     pub tags: Vec<String>,
 }
 
+impl  Message{
+    pub fn new_message(value: Vec<u8>)->Self{
+        Message{
+            value,
+            ..Default::default() // Use default values for keys and tags
+        }
+    }
+
+    pub fn message_to_drop()->Self{
+        Message{
+            tags:vec![DROP.parse().unwrap()],
+            ..Default::default() // Use default values for keys and tags
+        }
+    }
+
+    pub fn with_keys(mut self,keys:Vec<String>)->  Self{
+        self.keys=keys;
+        self
+    }
+
+    pub fn with_tags(mut self,tags:Vec<String>)->  Self{
+        self.tags=tags;
+        self
+    }
+
+    pub fn keys(mut self) ->Vec<String>{
+        self.keys
+    }
+    pub fn value(mut self) ->Vec<u8>{
+        self.value
+    }
+
+    pub fn tags(mut self) ->Vec<String>{
+        self.tags
+    }
+
```

Comment:
I was wondering, since these are public fields, do we really need getters and setters? I like the idea of the builder pattern because these fields are optional. Should we make the fields private instead?

Reply:
@BulkBeing WDYT on builder pattern? It is getting popular in Rust.